### ISSUE TITLE
Virtual property and load strategy

### ISSUE THREAD

**Issue by Arcus16** [mikro-orm/mikro-orm#3820]
**Describe the bug**
Virtual property is not populated for loadStrategy `LoadStrategy.JOINED`, but it works for `LoadStrategy.SELECT_IN`.

**To Reproduce**
Let's have a user and account entities with M:N relation, and a parent account modeled as M:1 relation. An additional requirement is that we want to access parentId for reading purposes.
The entities:
```typescript
@Entity()
export class Account extends BaseEntity<Account, 'id'> {
    @PrimaryKey()
    id!: number;

    @ManyToOne(() => Account, { nullable: true, joinColumn: 'parent_id' })
    parent!: Account | null;

    @Property({ persist: false })
    parentId!: number | null;

    @ManyToMany(() => User, user => user.accounts)
    users = new Collection<User>(this);
}

@Entity()
export class User extends BaseEntity<User, 'id'> {
    @PrimaryKey()
    id!: number;

    @ManyToMany(() => Account, 'users', {
        owner: true,
    })
    accounts = new Collection<Account>(this);
}
```
If we run following query:
```typescript
await this.em.findOne(User, { id: 11 }, { populate: ['accounts'] });
```
The resulting Account/s will differ based on the used load strategy. For `LoadStrategy.SELECT_IN` the Account will have property `parentId` populated, but not when `LoadStrategy.JOINED` strategy is used.

**Expected behavior**
Virtual property should populate also for `LoadStrategy.JOINED`

**Versions**

| Dependency | Version |
| - | - |
| node | 16.5.1|
| typescript | 4.4.3 |
| mikro-orm | 5.5.3 |
| mysql | 2.18.1 |

**Comment by B4nan** [comment]
FYI the fact that this works with select-in strategy is rather a side effect than a feature.

**Comment by Arcus16** [comment]
Ok, to make it work with the joined strategy I have only two options use a getter or serializer am I correct?

**Comment by B4nan** [comment]
Maybe we could bend the hydration layer to support this. One of the things I want to revisit in v6 is being able to do this without virtual properties - to be able to map one column to multiple entity properties.

**Comment by B4nan** [comment]
Fixed via d0b3698eebbbd2a74809b70f7aa37c82cee0359e
